daemon/daemon.go

```diff
@@ -310,6 +310,8 @@ func (daemon *Daemon) restore() error {
 				}
 			}
 
+			// reset container restart count
+			c.RestartManager(true)
```

Comment:
This is because we used two variables to calculate ShouldRestart: failureCount and RestartPolicy.

OK, I'd better merge these two params again, have had done it before, I'll try to migrate that part into this.